bin/pre-commit-hook.js

```diff
@@ -125,37 +127,232 @@ if ( phpLintResult && phpLintResult.status ) {
+
+		if ( ! file || ! rangesString ) {
+			return changedLines;
+		}
+
+		const ranges = rangesString.slice( 0, -1 ).split( ',' );
+		const lines = ranges.reduce( ( lines, range ) => {
+			const [ first, last ] = range.split( '-' ).map( num => parseInt( num, 10 ) );
+
+			let num = first;
+			do {
+				lines = [...lines, num++ ];
+			} while ( num <= last );
+
+			return lines;
+		}, [] );
+
+		return {...changedLines, [file]: lines };
+	}, {} );
 }
 
-if ( phpcbfResult && phpcbfResult.status ) {
-	execSync( `git add ${ phpcsFiles.join( ' ' ) }` );
-	console.log( chalk.yellow( 'PHPCS issues detected and automatically fixed via PHPCBF.' ) );
+/**
+ * Generates a PHPCS report from a PHPCS JSON blob
+ *
+ * @param {Object} PHPCS JSON Blob as generated, for example, by phpcs --report=json
+ */
+function phpcsReport( phpcsResults ) {
+	const table = require( 'table' );
+
+	const border = {
+		...table.getBorderCharacters( 'void' ),
+		topBody: '-', bottomBody: '-', joinBody: '-',
+		bodyJoin: '|',
+	};
+
+	const bodyConfig = {
+		border,
+		drawHorizontalLine: ( index, size ) => false,
+		columns: {
+			0: {
+				alignment: 'right',
+			}
+		}
+	}
+
+	for ( let [ file, { messages, errors, warnings, fixable } ] of Object.entries( phpcsResults.files ) ) {
+		if ( ! messages.length ) {
+			continue;
+		}
+
+		const lines = ( new Set( messages.map( message => message.line ) ) ).size;
+
+		const width = process.stdout.columns || 80;
+		const hr = '-'.repeat( width );
+		console.log( hr );
+		console.log( chalk.bold( `FILE: ${file}` ) );
+		console.log( hr );
+		console.log( chalk.bold( `FOUND ${errors} ERRORS AND ${warnings} WARNINGS AFFECTING ${lines} LINES` ) );
+		console.log( hr );
+
+		const body = messages.map( message => {
+			return [
+				message.line,
+				chalk['ERROR' === message.type ? 'red' : 'yellow']( message.type ),
+				`[${ message.fixable ? 'x' : ' ' }] ${chalk.bold( message.message )} (${message.source})`
+			];
+		} );
+
+		console.log( table.table( body, bodyConfig ).slice( 0, -1 ) );
+		console.log( hr )
+
+		if ( fixable ) {
+			console.log( chalk.bold( `PHPCBF CAN FIX THE ${fixable} MARKED SNIFF VIOLATIONS AUTOMATICALLY` ) );
+			console.log( hr + '\n' );
+		} else {
+			console.log();
+		}
+	}
 }
 
-if ( phpcsResult && phpcsResult.status ) {
-	const phpcsStatus = ( 2 === phpcsResult.status ? 'PHPCS reported some problems and could not automatically fix them since there are unstaged changes in the file.\n' : 'PHPCS reported some problems and cannot automatically fix them.\n' );
-	console.log(
-		chalk.red( 'COMMIT ABORTED:' ),
-			phpcsStatus +
-			'If you are aware of them and it is OK, ' +
-			'repeat the commit command with --no-verify to avoid this check.\n' +
-			"But please don't. Code is poetry."
-	);
-	exitCode = 1;
+/**
+ * Lints files and filters the report to only worry about errors that occur on changed lines.
+ *
+ * @param {Array} files
+ */
+function phpcsLinter( files ) {
```

Comment:
Consider outputting each state changing command that runs so that if something goes wrong, at least you can see what it did.